elastic/kibana
# [Uptime] Duration chart gaps

## Summary

Resolves https://github.com/elastic/uptime/issues/54.

This patch adds gaps to our duration chart when there is no data for a location, which will cause the series to more accurately reflect the state of the monitor if it is down.

<img width="1152" alt="Screen Shot 2019-09-13 at 6 59 49 PM" src="https://user-images.githubusercontent.com/18429259/64900098-e9d69400-d65c-11e9-93d0-912fae54d94a.png">

## Testing this PR
0. Configure a web service you can control that will return a `200`. Set up a stack running this patch and begin pinging your service with Heartbeat.
1. Load this patch, navigate to your service's _Monitor_ page.
2. After accumulating enough data to begin seeing line series for your locations, kill your service. Allow this `down` data to accumulate.
3. After enough time has passed for some empty buckets to appear for your selected date range, start your service back up again.
4. Observe that you have empty gaps for the periods of time that your monitor was `down`.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.


### For maintainers
